test/PositionManager.ts

```diff
@@ -131,10 +191,201 @@ describe("GammaPoolFactory", function () {
                 expect(event.args.pool).to.equal(gammaPool.address);
                 expect(event.args.assets.toNumber()).to.equal(16);
             });
-        });/**/
+        });
+
+        it("depositReserves", async function () {
+            await cfmm.approve(posMgr.address, ethers.constants.MaxUint256);//must approve before sending tokens
+            const DepositReservesParams =  {
+                cfmm: cfmm.address,
+                amountsDesired: [10000, 100],
+                amountsMin: [1000, 10],
+                to: addr4.address,
+                protocol: 1,
+                deadline: ethers.constants.MaxUint256
+            }
+            const res = await (await posMgr.depositReserves(DepositReservesParams)).wait();
+            res.events.forEach(function(event: any, i: any){
+                expect(event.args).to.not.be.an('undefined');
+                console.log("event >>");
+                console.log(event.args);
+                expect(event.args.pool).to.equal(gammaPool.address);
+                expect(event.args.reservesLen).to.equal(4);
+                expect(event.args.shares.toNumber()).to.equal(18);
+            });
+        });
+
+        it("withdrawReserves", async function () {            
+            await gammaPool.approve(posMgr.address, ethers.constants.MaxUint256);//must approve before sending tokens
+
+            const WithdrawReservesParams =  {
+                cfmm: cfmm.address,
+                protocol: 1,
+                amount: 1000,
+                amountsMin: [100, 200, 300],
+                to: owner.address,
+                deadline: ethers.constants.MaxUint256
+            }
+
+            const res = await (await posMgr.withdrawReserves(WithdrawReservesParams)).wait();
+
+            res.events.forEach(function(event: any, i: any){
+                if(i == 0)
+                    return;
+                console.log("event >>");
+                console.log(event.args);
+                expect(event.args.pool).to.equal(gammaPool.address);
+                expect(event.args.reservesLen).to.equal(3);
+                expect(event.args.assets.toNumber()).to.equal(17);
+            });
+        });
     });
+
     // You can nest describe calls to create subsections.
     describe("Long Gamma", function () {
+        it("createLoan", async function () {
+            await cfmm.approve(posMgr.address, ethers.constants.MaxUint256);//must approve before sending tokens
+            
```

Comment:
Should create the loan here not create it in the beforeEach statement

Reply:
I have created loan in the before each, cause its tokenId is needed throughout the testcases, hardcoded is not working.

Reply:
how are you hardcoding the tokenId?